packages/insomnia-app/app/models/helpers/__tests__/settings.test.ts

```diff
@@ -0,0 +1,208 @@
+      value: true,
+    });
+  });
+});
+
+describe('omitControlledSettings', () => {
+  it('omits config controlled settings', () => {
+    getConfigSettings.mockReturnValue({ disablePaidFeatureAds: true });
+    const settings = models.settings.init();
+
+    const result = omitControlledSettings(settings)({ disablePaidFeatureAds: false });
+
+    expect(result).toMatchObject({});
+  });
+
+  it('does not omit settings not controlled by the config', () => {
+    getConfigSettings.mockReturnValue({});
+    const settings = models.settings.init();
+
+    const result = omitControlledSettings(settings)({ disablePaidFeatureAds: true });
+
+    expect(result).toMatchObject({ disablePaidFeatureAds: true });
+  });
+
+  it('omits settings controlled by other settings', () => {
+    getConfigSettings.mockReturnValue({});
+    const settings: Settings = {
+      ...models.settings.init(),
+      incognitoMode: true,
+    };
+    const result = omitControlledSettings(settings)({ enableAnalytics: true });
+
+    expect(result).toMatchObject({});
+  });
+
+  it('does not omit settings not controlled by other settings', () => {
+    getConfigSettings.mockReturnValue({});
+    const settings = models.settings.init();
+    const result = omitControlledSettings(settings)({ disablePaidFeatureAds: true });
+
+    expect(result).toMatchObject({ disablePaidFeatureAds: true });
+  });
+});
+
+describe('getControlledSettings', () => {
+  it('overwrites config controlled settings', () => {
+    getConfigSettings.mockReturnValue({ disablePaidFeatureAds: true });
+    const settings: Settings = {
+      ...models.settings.init(),
+      disablePaidFeatureAds: false,
+    };
+
+    const result = getControlledSettings(settings);
+
+    expect(result).toMatchObject({ disablePaidFeatureAds: true });
+  });
+
+  it('does not overwrite settings not controlled by the config', () => {
+    getConfigSettings.mockReturnValue({});
+    const settings: Settings = {
+      ...models.settings.init(),
+      disablePaidFeatureAds: true,
+    };
+
+    const result = getControlledSettings(settings);
+
+    expect(result).toMatchObject({ disablePaidFeatureAds: true });
+  });
+
+  it('overwrites settings controlled by other settings', () => {
+    getConfigSettings.mockReturnValue({});
+    const settings: Settings = {
+      ...models.settings.init(),
+      incognitoMode: true,
+      enableAnalytics: true,
+    };
+
+    const result = getControlledSettings(settings);
+
+    expect(result).toMatchObject({ enableAnalytics: false });
+  });
+
+  it('does not overwrite settings not controlled by other settings', () => {
+    getConfigSettings.mockReturnValue({});
+    const settings: Settings = {
+      ...models.settings.init(),
+      disablePaidFeatureAds: true,
+    };
+
+    const result = getControlledSettings(settings);
+
+    expect(result).toMatchObject({ disablePaidFeatureAds: true });
+  });
+
+  it.skip('config control trumps (non-config) settings control', () => {
```

Comment:
Intentionally skipped?

```suggestion
  it.skip('prioritizes config file settings above all else', () => {
```

Reply:
thanks, updated in 84ec07d4e3b2d1acba73d4b7240216c0441ce9f8

Reply:
intentionally skipped, yes, and should be removed.  thanks.  removed in cdf9a85aceab36072c5a452664069c43016597e4